runtime/doc/lua.txt

```diff
@@ -3388,6 +3388,29 @@ Iter:find({f})                                                   *Iter:find()*
     Return: ~
         any
 
+Iter:flatten({depth})                                         *Iter:flatten()*
+    Flattens a |list-iterator| pipeline, un-nesting nested values. The
+    flattened list will exclude nil, and any values with non-numeric keys
```

Comment:
This note seems misleading. Can a `|list-iterator|` ever have a non-numeric key? Or is `|list-iterator|` only required at the "top level" (and then next question would be "why")?

Reply:
This is now required at every level